azure-iot-device/azure/iot/device/common/models/x509.py

```diff
@@ -0,0 +1,39 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module represents a certificate that is responsible for providing client provided x509 certificates
+that will eventually establish the authenticity of devices to IotHub and Provisioning Services.
+"""
+
+
+class X509(object):
+    """
+    A class with references to the certificate, key, and optional pass-phrase used to authenticate
+    a TLS connection using x509 certificates
+    """
+
+    def __init__(self, cert_file, key_file, pass_phrase=None):
+        """
+        Initializer for X509 Certificate
+        :param cert_file: The file path to contents of the certificate (or certificate chain)
+         used to authenticate the device.
+        :param key_file: The file path to the key associated with the certificate
+        :param pass_phrase: (optional) The pass_phrase used to encode the key file
+        """
+        self._cert = cert_file
+        self._key = key_file
+        self._pass_phrase = pass_phrase
+
+    @property
+    def certificate(self):
```

Comment:
>certificate [](start = 8, length = 11)

nit: SSLContext uses "certfile" and "keyfile" and i think we should do the same just to make it very clear
there has been some confusion in the node sdk between the certificate *file* and its content, it should be very explicit from the user that we expect a file path and not the content of the file itself.

Reply:
i don't see docstrings on properties. 

---
In reply to: [293189715](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r293189715) [](ancestors = 293189715)